docassemble/ALToolbox/share_button.py

```diff
@@ -0,0 +1,30 @@
+from docassemble.base.functions import word, log
+
+# See GitHub issue https://github.com/SuffolkLITLab/docassemble-ALToolbox/issues/16
+def share_button (
+  text_to_copy,
+  text_before='',
+  label='Copy',
+  tooltip_inert_text='Copy to clipboard',
+  tooltip_copied_text='Copied: '  # Will have `text_to_copy` added to it
+):
+  '''Return the html for a button that will let a user share the given text'''
+  button_str = '<div class="al_share">\n'
+  if ( text_before != '' ):
+    button_str += f'<span class="al_share_description">{word( text_before )}</span>\n'
+  button_str += f'<input readonly class="al_share_value" type="text" value="{word( text_to_copy )}">\n'
+  button_str += f'<button class="btn btn-primary al_share_button" type="button">\n'
+  button_str += f'<span class="al_tooltip al_tooltip_inert">{word( tooltip_inert_text )}</span>\n'
+  
+  if ( tooltip_copied_text == 'Copied: ' ):
+    # Our default text will show 'Copied: text that will be copied'
+    button_str += f'<span class="al_tooltip al_tooltip_active">Copied: {word( tooltip_copied_text ) + text_to_copy }</span>\n'
+  else:
+    # Otherwise, use the custom text handed in
+    button_str += f'<span class="al_tooltip al_tooltip_active">Copied: {word( tooltip_copied_text )}</span>\n'
```

Comment:
This is confusing to me. In both cases it says "Copied" without any translation. And what is the distinction between the two branches? Shouldn't it just look like 
`button_str += f'<span class="al_tooltip al_tooltip_active">{word( tooltip_copied_text ) + text_to_copy }</span>\n'`
in both described circumstances? An option to not show the copied text seems reasonable, but this method of invoking it doesn't make sense to me.

Reply:
Would the option to not show the copied text be another parameter? If I were a dev using this module and handed in something to show in the tooltip called 'tooltip_copied_text', I would expect the value I give to be the one shown in the tooltip after the text had been copied. That is, I wouldn't expect it to be modified.

We could split it up into two parameters perhaps - one for the text before the copied text and one for choosing to not show the copied text. I haven't been able to think of a clear name for the first param, though. `tooltip_text_to_the_left_of_copied_text`? The other could be something like `show_copied_text_in_tooltip` or something.

Reply:
IMO, you don't need to show the copied text in the tooltip at all? It's already on the screen and associated with the button. And when the tooltip is really long, the beginning just cuts off, which is much more confusing

![Screenshot from 2021-05-17 09-36-21](https://user-images.githubusercontent.com/6252212/118497836-70c1f680-b6f3-11eb-8f47-64f6945a70b2.png)

For names, I think the existing `tooltip_copied_text` is great, and doesn't need to change, if we do want to include the copied text in the tooltip, a good name would be `append_copied_text_to_tooltip`

Reply:
I'd be fine with removing the text that actually gets added to the clipboard from the "Copied!" tooltip. That would simplify this.